system-image-upgrader

```diff
@@ -501,6 +501,12 @@ do
             xzcat recovery/$2 | tar --numeric-owner -xf -
             rm -f removed
 
+            # Check for legacy channel flag and hardlink system.img again
+            if [ -e legacy_channel ]; then
+                ln /data/ubuntu.img /data/system.img || true
+                rm legacy_channel
```

Comment:
It appears that all other `rm` commands in this file are `rm -f`, please replicate.